### PR TITLE
GEOMESA-1854 Replacing ciFlatMap with flatMap in CloseableIterator

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/collection/CloseableIterator.scala [hunk 1]
```diff
@@ -161,12 +155,12 @@ object SelfClosingIterator {
         iterHasNext
       }
       override def next(): A = iter.next()
-      override def close(): Unit = closeIter()
+      override def close(): Unit = closeIter
     }
 
-  def apply[A](iter: Iterator[A] with Closeable): SelfClosingIterator[A] = apply(iter, iter.close)
+  def apply[A](iter: Iterator[A] with Closeable): SelfClosingIterator[A] = apply(iter, iter.close())
```
Comment (jnh5y) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/collection/CloseableIterator.scala]: Hmm.... does this call iter.close() on construction?

Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/collection/CloseableIterator.scala]: no, pretty sure not - i changed the apply syntax from `() => Unit` to `=> Unit` so the syntax of calling it is slightly different. i'll add a unit test just to make sure

Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/collection/CloseableIterator.scala]: added

Comment (jnh5y) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/collection/CloseableIterator.scala]: Ah, ok, thanks for the unit test.  I missed the both old and new versions are 'productions' with =>.  I feel better now:).

**Pull request merged**